### PR TITLE
Allow additional OIDC domains to access APIs when OAuth is used

### LINKED ISSUE #2420: Allow multiple OIDC domains to be authenticated when calling APIs in OAuth mode
Allow multiple OIDC domains to be authenticated when calling APIs in OAuth mode. Add a new list of allowed domains to server config and change the login verification to determine if the email in the ID token belongs to any of the allowed domains.

### PR DESCRIPTION
**Pull request by sa3eed3ed**
Allow multiple OIDC domains to be authenticated when calling APIs in OAuth mode. Add a new list of allowed domains to server config and change the login verification to determine if the email in the ID token belongs to any of the allowed domains.

closes #2420

### CODE REVIEW

File: timesketch/views/auth.py [hunk 1]
```diff
@@ -297,10 +297,19 @@ def validate_api_token():
...
     # Check if the authenticating user is part of the allowed domains.
+    allowed_domains = set()
     hosted_domains = current_app.config.get("GOOGLE_OIDC_HOSTED_DOMAIN")
+    api_allowed_domains = current_app.config.get("GOOGLE_OIDC_API_ALLOWED_DOMAINS", [])
     if hosted_domains:
+        allowed_domains.add(hosted_domains)
+    if api_allowed_domains:
+        allowed_domains.update(api_allowed_domains)
+    # A list of allowed domains in lower case.
+    ALLOWED_DOMAINS = list(map(lambda x: x.lower(), allowed_domains))
```
Comment (berggren) [inline timesketch/views/auth.py:308]: To stay consistent with the style in the project, we favour list comprehensions over lambdas when possible.
 ```suggestion
    ALLOWED_DOMAINS = [domain.lower() for domain in allowed_domains]
```

**Review by berggren — CHANGES_REQUESTED** [review]

File: config/dpkg/timesketch-server.timesketch.default [hunk 2]
```diff
@@ -107,6 +107,9 @@ GOOGLE_OIDC_CLIENT_SECRET = None
 # Limit access to a specific Google GSuite domain.
 GOOGLE_OIDC_HOSTED_DOMAIN = None
 
+# Additional Google GSuite domains allowed API access.
+GOOGLE_OIDC_API_ALLOWED_DOMAINS = None
```
Comment (sa3eed3ed) [inline config/dpkg/timesketch-server.timesketch.default:111]:
```suggestion
GOOGLE_OIDC_API_ALLOWED_DOMAINS = []
```

File: timesketch/views/auth.py [hunk 3]
```diff
@@ -297,10 +297,19 @@ def validate_api_token():
...
     # Check if the authenticating user is part of the allowed domains.
+    allowed_domains = set()
     hosted_domains = current_app.config.get("GOOGLE_OIDC_HOSTED_DOMAIN")
+    api_allowed_domains = current_app.config.get("GOOGLE_OIDC_API_ALLOWED_DOMAINS", [])
     if hosted_domains:
+        allowed_domains.add(hosted_domains)
+    if api_allowed_domains:
+        allowed_domains.update(api_allowed_domains)
+    # A list of allowed domains in lower case.
+   ALLOWED_DOMAINS = [domain.lower() for domain in allowed_domains]
```
Comment (sa3eed3ed) [inline timesketch/views/auth.py:308]:
```suggestion
    ALLOWED_DOMAINS = [domain.lower() for domain in allowed_domains]
```

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**